src/main/java/dukchess/dao/TaskList.java

```diff
@@ -0,0 +1,78 @@
+package dukchess.dao;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import dukchess.entity.Deadline;
+import dukchess.entity.Event;
+import dukchess.entity.Task;
+import dukchess.entity.Todo;
+
+public class TaskList extends ArrayList<Task> {
+    private File taskFile;
+    private FileWriter taskFileWriter;
+    private Scanner taskFileScanner;
+
+    public TaskList(File taskFile) throws IOException {
+        super();
+        this.taskFile = taskFile;
+        this.taskFileScanner = new Scanner(taskFile);
+        initializeFromFile();
```

Comment:
The details of this method was abstracted, but it is trying initialize something, on top of the constructor. You could consider if [SLAP](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard) could be possible here.